### PR TITLE
[exporter/awsemf] Reuse pushing logic

### PR DESCRIPTION
**Pull request by aateeqi**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6991
**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>
PR based on discussion in 
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6969

### CODE REVIEW

File: internal/aws/cloudwatch/go.mod [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cloudwatch
```
Comment (anuraaga) [inline internal/aws/cloudwatch/go.mod:1]: I think this module should be called `cloudwatchlogs`, or alternatively maybe have a `logs` package within `cloudwatch` but lean towards the former given how AWS documentation also has both as top level types

https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/Welcome.html
https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/Welcome.html

Anyways, just `cloudwatch` being logs code is misleading since Amazon CloudWatch is a metrics service, not a logs service

File: exporter/awsemfexporter/grouped_metric_test.go [hunk 2]
```diff
@@ -33,6 +33,11 @@ import (
 
 	internaldata "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus"
 )
+var previousSequenceToken = "0000"
```
Comment (anuraaga) [inline exporter/awsemfexporter/grouped_metric_test.go:36]:
```suggestion

var previousSequenceToken = "0000"
```

File: go.mod [hunk 3]
```diff
@@ -115,6 +115,11 @@ require (
 	golang.org/x/sys v0.0.0-20211210111614-af8b64212486
 )
 
+require (
```
Comment (anuraaga) [inline go.mod:118]: Was this change automated? It looks unrelated to this PR

File: internal/aws/cloudwatch/handler/request_structuredlog_handler.go [hunk 4]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package handler // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/handler"
```
Comment (anuraaga) [inline internal/aws/cloudwatch/handler/request_structuredlog_handler.go:15]: Looks like we lost the canonical import paths in the refactoring (TBH I don't know why we need to write such boring code though)

File: testbed/go.mod [hunk 5]
```diff
@@ -33,6 +33,11 @@ require (
 	golang.org/x/text v0.3.7
 )
 
+require (
```
Comment (anuraaga) [inline testbed/go.mod:36]: Revert this file

File: internal/aws/cloudwatch/go.mod (hunk 1 above)
Comment (aateeqi) [reply internal/aws/cloudwatch/go.mod:1]: @anuraaga so this package is going to be used to push cloudwatch metrics and also logs, so wouldn't it be appropriate to call it cloudwatch?

Comment (anuraaga) [reply internal/aws/cloudwatch/go.mod:1]: The metrics are pushed via logs - this package I think is intended only for logs for now. Either way logs and metrics are too different to be in the same package, I feel the API docs make that clear.

File: internal/aws/cwlogs/pusher.go [hunk 6]
```diff
@@ -47,37 +47,37 @@ var (
 )
 
 // logEvent struct to present a log event.
-type logEvent struct {
-	inputLogEvent *cloudwatchlogs.InputLogEvent
+type LogEvent struct {
```
Comment (Aneurysm9) [inline internal/aws/cwlogs/pusher.go:50]: This can just be `Event`.

File: internal/aws/cwlogs/pusher.go [hunk 7]
```diff
@@ -47,37 +47,37 @@ var (
...
+	InputLogEvent *cloudwatchlogs.InputLogEvent
 	// The time which log generated.
-	logGeneratedTime time.Time
+	LogGeneratedTime time.Time
```
Comment (Aneurysm9) [inline internal/aws/cwlogs/pusher.go:53]: Since there is no other time field here and we know we're dealing with logs, would this be better as `GeneratedTime`?

File: internal/aws/cwlogs/pusher.go [hunk 8]
```diff
@@ -146,14 +146,14 @@ func (batch *logEventBatch) isActive(targetTimestampMs *int64) bool {
 	return true
 }
 
-func (batch *logEventBatch) append(event *logEvent) {
-	batch.putLogEventsInput.LogEvents = append(batch.putLogEventsInput.LogEvents, event.inputLogEvent)
+func (batch *logEventBatch) append(event *LogEvent) {
```
Comment (Aneurysm9) [inline internal/aws/cwlogs/pusher.go:149]: Perhaps less significant, but maybe `eventBatch` here?  Basically, strip `log` from everywhere that it is repetitive or unnecessary since the whole purpose of this package is to handle logs.

File: exporter/awsemfexporter/emf_exporter.go [hunk 9]
```diff
@@ -33,6 +33,7 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/awsutil"
+	cloudwatch "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs"
```
Comment (anuraaga) [inline exporter/awsemfexporter/emf_exporter.go:36]: Let's remove the alias throughout the PR, so below things look like `cwlogs.Pusher` etc

**Review by anuraaga — APPROVED** [review]
Thanks!

**Comment by jpkrohling** [comment]
Is this PR closing the linked issue? Or is just a step in that direction?

**Comment by aateeqi** [comment]
> Is this PR closing the linked issue? Or is it just a step in that direction?

Hey @jpkrohling , yes this would just be a step in allowing `awscloudwatchlogsexporter` to use the new package

**Review by Aneurysm9 — APPROVED** [review]
